### PR TITLE
Replace xero-oss/go-xray with jasonwbarnett/go-xray and add repo_type to Xray Watch

### PR DESCRIPTION
**Pull request by jasonwbarnett**
I have not yet tested this, but in theory it should just work. 😸 

This fixes https://github.com/jfrog/terraform-provider-artifactory/issues/56

### CODE REVIEW

**Comment by chb0github** [comment]
If you need a test instance, find me in gitter for details

**Comment by chb0github** [comment]
waiting on you

**Comment by jasonwbarnett** [comment]
@chb0github I completely tested this out finally and it's all working.

All CRUD operations of both local and remote repositories to Xray Watches.

File: docs/resources/artifactory_xray_watch.md [hunk 1]
```diff
@@ -37,6 +37,7 @@ The top-level `resources` block contains a list of one or more resource objects
 * `type` - (Required) Type of resource to be watched
 * `name` - (Required) A name describing the resource
 * `bin_mgr_id` - (Optional) The ID number of a binary manager resource
+* `repo_type` - (Optional) Type of repository (e.g. local or remote)
```
Comment (chb0github) [inline docs/resources/artifactory_xray_watch.md:40]: Thank you for doc update

File: sample.tf [hunk 2]
```diff
@@ -37,41 +37,62 @@ resource "artifactory_local_repository" "npm-local" {
...
+      type  = "package-type"
+      value = "Npm"
+    }
+  }
+
+  resources {
+    type       = "repository"
+    name       = artifactory_remote_repository.npm-remote.key
+    bin_mgr_id = "default"
+    repo_type  = "remote"
```
Comment (chb0github) [inline sample.tf:86]: Thanks for updating the tf

**Review by chb0github — APPROVED** [review]
Looks good

**Pull request merged**